nidmviewer/viewer.py

```diff
@@ -92,6 +94,16 @@ def generate(ttl_files,base_image=None,retrieve=False,view_in_browser=False,colu
 
     if view_in_browser==True:
         peaks,copy_list = generate_temp(peaks,"excsetmap_location")
+        # Check if the nifti is empty
+        for exc_set_file in copy_list.keys():
+            img = nib.load(exc_set_file)
+            data = img.get_data()
+            data = np.nan_to_num(data)
+            if np.count_nonzero(data) == 0:
+                # Empty excursion set is removed from the display list
+                del copy_list[exc_set_file]
+                peaks[peaks.keys()[0]] = {}
+
         if base_image == None:
```

Comment:
I'm quite confused what this piece of code was supposed to solve. Let me make some general points (correct me if I'm wrong).

1. If there are not peaks listed in the .ttl file, but the excursion set nifti file is not empty show "No peak data available" in the table on the left and the excursion set nifti file on the right. (if I understand correctly this is what happens for all NIDM-Results files generated by FSL).
2. If there are no peaks listed in the .ttl file, and the excursion set nifti file **is empty** show "No suprathreshold voxels" instead of the table and the nifiti viewer. (this will happen if the analysis did not return any statistically significant results)
3. For other maps (where there is info about peak coordinates) each peak coordinate should correspond to a cluster in the excursion set nifti file. No filtering of peak coordinates should be necessary. If this is not the case something is wrong (although I would not rule out problems with the test data).

I hope this helps!